src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/cache_tuning/FieldDataCacheDeciderITest.java

```diff
@@ -0,0 +1,255 @@
+                                        sum = 100.0, avg = 100.0, min = 100.0, max = 100.0)
+                        })
+        })
+@AMetric(
+        name = Heap_Max.class,
+        dimensionNames = {AllMetrics.HeapDimension.Constants.TYPE_VALUE},
+        tables = {
+                @ATable(
+                        hostTag = HostTag.DATA_0,
+                        tuple = {
+                                @ATuple(
+                                        dimensionValues = {AllMetrics.GCType.Constants.HEAP_VALUE},
+                                        sum = 1000000.0, avg = 1000000.0, min = 1000000.0, max = 1000000.0),
+                                @ATuple(
+                                        dimensionValues = {AllMetrics.GCType.Constants.HEAP_VALUE},
+                                        sum = 1000000.0, avg = 1000000.0, min = 1000000.0, max = 1000000.0)
+                        }),
+                @ATable(
+                        hostTag = {HostTag.ELECTED_MASTER},
+                        tuple = {
+                                @ATuple(
+                                        dimensionValues = {AllMetrics.GCType.Constants.HEAP_VALUE},
+                                        sum = 1000000.0, avg = 1000000.0, min = 1000000.0, max = 1000000.0),
+                                @ATuple(
+                                        dimensionValues = {AllMetrics.GCType.Constants.HEAP_VALUE},
+                                        sum = 1000000.0, avg = 1000000.0, min = 1000000.0, max = 1000000.0)
+                        })
+        })
+
+public class FieldDataCacheDeciderITest {
+    public static final String CACHE_TUNING_RESOURCES_DIR = Consts.INTEG_TESTS_SRC_DIR + "./tests/cache_tuning/resource/";
+    public static final String INDEX_NAME = "MockIndex";
+    public static final String SHARD_ID = "1";
+
+    // Test CacheDecider for ModifyCacheAction (field data cache).
+    // The cache decider should emit modify cache size action as field data rca is unhealthy.
+    @Test
+    @AExpect(
+            what = AExpect.Type.DB_QUERY,
+            on = HostTag.ELECTED_MASTER,
+            validator = FieldDataCacheDeciderValidator.class,
+            forRca = PersistedAction.class,
+            timeoutSeconds = 1000)
+    @AErrorPatternIgnored(
+            pattern = "AggregateMetric:gather()",
+            reason = "CPU metrics are expected to be missing in this integ test")
+    @AErrorPatternIgnored(
+            pattern = "Metric:gather()",
+            reason = "Metrics are expected to be missing in this integ test")
+    @AErrorPatternIgnored(
+            pattern = "NodeConfigCacheReaderUtil",
+            reason = "Node Config Cache are expected to be missing in this integ test.")
+    @AErrorPatternIgnored(
+            pattern = "CacheUtil:getCacheMaxSize()",
+            reason = "Node Config Cache is expected to be missing during startup.")
```

Comment:
If we add the relevant metric values for them, will these caches get populated. By skipping this error in integ tests we expose ourselves to these failures in actual clusters.

Reply:
Right that was the thought behind adding all cache metrics to ITest, the ability to catch these. Removing them

Reply:
I have removed un-wanted metrics (gives us a clearer understanding of what metrics are related to the decider). Having these errors. In case we get these errors, our validations will eventually fail.